Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/Curvatures/interpolated_corrected_curvature_measures.h

```diff
@@ -0,0 +1,973 @@
+// Copyright (c) 2022 GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     : Hossam Saeed
+//
+
+#ifndef CGAL_POLYGON_MESH_PROCESSING_INTERPOLATED_CORRECTED_CURVATURE_MEASURES_H
+#define CGAL_POLYGON_MESH_PROCESSING_INTERPOLATED_CORRECTED_CURVATURE_MEASURES_H
+
+#include <CGAL/license/Polygon_mesh_processing/interpolated_corrected_curvature_measures.h>
+
+#include <CGAL/assertions.h>
+#include <CGAL/Polygon_mesh_processing/internal/named_params_helper.h>
+#include <CGAL/Polygon_mesh_processing/compute_normal.h>
+#include <CGAL/Named_function_parameters.h>
+#include <CGAL/property_map.h>
+#include <Eigen/Eigenvalues>
+
+#include <numeric>
+#include <queue>
+#include <unordered_set>
+#include <functional>
+
+namespace CGAL {
+
+namespace Polygon_mesh_processing {
+
+/*!
+ * \ingroup PMP_corrected_curvatures_grp
+ * Enumeration type used to specify which measure of a given face
+ * is computed for the interpolated corrected curvature functions
+ */
+// enum
+enum Curvature_measure_index {
+    MU0_AREA_MEASURE, ///< corrected area density
+    MU1_MEAN_CURVATURE_MEASURE, ///< corrected mean curvature density
+    MU2_GAUSSIAN_CURVATURE_MEASURE ///< corrected gaussian curvature density
+};
+
+/**
+* \ingroup PMP_corrected_curvatures_grp
+*
+* computes the interpolated corrected area measure (mu0) of a specific face.
+*
+* @tparam GT is the geometric traits class.
+*
+* @param x is a vector of the vertex positions of the face.
+* @param u is a vector of the vertex nomrals of the face.
+*
+* @return a scalar of type `GT::FT`.
+* This is the value of the interpolated corrected area measure of the given face.
+*
+* @see `interpolated_corrected_mean_curvature_measure_face()`
+* @see `interpolated_corrected_gaussian_curvature_measure_face()`
+* @see `interpolated_corrected_measure_mesh()`
+*/
+template<typename GT>
+typename GT::FT interpolated_corrected_area_measure_face(const std::vector<typename GT::Vector_3>& u,
```

Comment:
@sloriot, it seems like it would be difficult to use `Point_3`, as in multiple occasions it is needed to take cross product with a `Vector_3`. More over, in a lot of cases it is also needed to sum some `Points` with weights that add up to `0`. In that case I can't use `CGAL::barycenter()`. What would you recommend?